File: 189-rotate-array/rotate-array.ts
/**
 Do not return anything, modify nums in-place instead.
 */
function rotate(nums: number[], k: number): void {
    let ans = [];
    let n = nums.length;
    for (let i = 0; i < n; i++) {
        ans[(i + k) % n] = nums[i];
    }
    for (let i = 0; i < n; i++) nums[i] = ans[i];
};

File: 189-rotate-array/rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        int[] helper = new int[nums.length];
        for(int i = 0; i < nums.length; i ++) {
            helper[(i + k) % nums.length] = nums[i];
        }
        for(int i = 0; i < nums.length; i ++){
            nums[i] = helper[i];
        }
    }
}